mailpoet/assets/js/src/automation/editor/components/panel/activate-panel.tsx

```diff
@@ -0,0 +1,114 @@
+import { StoreDescriptor, useDispatch, useSelect } from '@wordpress/data';
+import { Button } from '@wordpress/components';
+import { store as noticesStore } from '@wordpress/notices';
+import { closeSmall } from '@wordpress/icons';
+import { __ } from '@wordpress/i18n';
+import { storeName } from '../../store';
+import { WorkflowStatus } from '../../../listing/workflow';
+import { MailPoet } from '../../../../mailpoet';
+
+function PreStep({ onClose }): JSX.Element {
+  const { activate } = useDispatch(storeName);
+
+  return (
+    <>
+      <div className="mailpoet-automation-activate-panel__header">
+        <div className="mailpoet-automation-activate-panel__header-activate-button">
+          <Button
+            variant="primary"
+            onClick={() => {
+              activate();
+            }}
+          >
+            {__('Activate', 'mailpoet')}
+          </Button>
+        </div>
+
+        <div className="mailpoet-automation-activate-panel__header-cancel-button">
+          <Button variant="secondary" onClick={onClose}>
+            {__('Cancel', 'mailpoet')}
+          </Button>
+        </div>
+      </div>
+
+      <div className="mailpoet-automation-activate-panel__body">
+        <p>
+          <strong>{__('Are you ready to activate?', 'mailpoet')}</strong>
+        </p>
+        <p>{__('Double-check your settings before activating', 'mailpoet')}</p>
+      </div>
+    </>
+  );
+}
+
+function PostStep({ onClose }): JSX.Element {
+  const { workflow } = useSelect(
+    (select) => ({
+      workflow: select(storeName).getWorkflowData(),
+    }),
+    [],
+  );
+
+  const goToListings = () => {
+    window.location.href = MailPoet.urls.automationListing;
+  };
+
+  const { createErrorNotice } = useDispatch(noticesStore as StoreDescriptor);
+  void createErrorNotice(
```

Comment:
There should be no business logic in the component body (that can re-render any number of times). This is a "side-effect" that causes other rendering so it results in the following error:

<img width="776" alt="Screen Shot 2022-10-19 at 7 24 57" src="https://user-images.githubusercontent.com/141436/196605101-84abcfe0-9960-4039-9339-1f4cc734b085.png">

If you want to display the notice on component mount, you can do so with `useEffect` with an empty deps array (well, almost empty — having only the dispatch action):

```ts
useEffect(() => {
  void createErrorNotice(...);
}, [createErrorNotice]);
```

(Alternatively, it could be triggered [from within the activate action](https://github.com/mailpoet/mailpoet/blob/3ea730a0b642b6c59875f0a0fec4e858d71315a7/mailpoet/assets/js/src/automation/editor/store/actions.ts#L81) right after the `yield`).

Reply:
I like the idea to update it in the `active` method itself, so we have a "universal" behavior in case we activate somewhere else in the future as well. Addressed in f5989e412